### PR TITLE
[CMAKE] Fix windows cmake build

### PR DESCRIPTION
**Pull request by tqchen**
The new NNVM source tree contains nnvm compiler which should not be included when compiling nnvm, update the source tree to only contain the nnvm core library

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -253,7 +253,10 @@ FILE(GLOB_RECURSE SOURCE "src/*.cc" "src/*.h")
 FILE(GLOB_RECURSE CUDA "src/*.cu" "src/*.cuh")
 
 # add nnvm to source
-FILE(GLOB_RECURSE NNVMSOURCE "nnvm/src/*.cc" "nnvm/src/*.h" "nnvm/include/*.h")
+FILE(GLOB_RECURSE NNVMSOURCE
+  nnvm/src/c_api/*.cc
```
Comment (cjolivier01) [inline CMakeLists.txt]: Looks ok. Can you keep the headers in?  This helps IDE's such as CLion know to index the file.

File: appveyor.yml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 version: 1.0.{build}
 
 environment:
-    MXNET_VS: "Visual Studio 12 2013 Win64"
+    MXNET_VS: "Visual Studio 14 2015 Win64"
```
Comment (piiswrong) [inline appveyor.yml]: Do we have to drop support for 2013?

Comment (tqchen) [reply appveyor.yml]: I can enable it by checking MSVC version seems was due to the introduction of advanced template here https://github.com/dmlc/nnvm/blob/master/include/nnvm/tuple.h#L615

**Comment by tqchen** [comment]
@piiswrong I have updated nnvm to enable msvc 13 build, seems the appveyor build for nnvm now pass

File: CMakeLists.txt (hunk 1 above)
Comment (tqchen) [reply CMakeLists.txt]: updated

**Comment by piiswrong** [comment]
Will this include conv2d etc into mxnet?

**Comment by tqchen** [comment]
those are in the source tree of nnvm/src/top and nnvm/src/compiler, and not part of the libnnvm core source tree, so this change won't include these additional defs into mxnet until we decided to do so

**Comment by tqchen** [comment]
@piiswrong @cjolivier01 this  now passes the tests

**Comment by cjolivier01** [comment]
You want this in v0.12.0?

**Comment by cjolivier01** [comment]
Rebased into v0.12.0

**Pull request merged**